app/src/main/java/com/example/diceroller/MainActivity.java

```diff
@@ -16,14 +16,73 @@
 
 package com.example.diceroller;
 
-import androidx.appcompat.app.AppCompatActivity;
 import android.os.Bundle;
+import android.widget.Button;
+import android.widget.ImageView;
+import androidx.appcompat.app.AppCompatActivity;
 
+/**
+ * This activity allows the user to roll a dice and view the result
+ * on the screen.
+ */
 public class MainActivity extends AppCompatActivity {
 
+    /**
+     * This method is called when the Activity is created.
+     */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        // Find the Button in the layout
+        Button rollButton = findViewById(R.id.button);
+
+        // Set a click listener on the button to roll the dice when the user taps the button
+        rollButton.setOnClickListener(v -> rollDice());
```

Comment:
Instead of this syntax, should we explicitly write it out all out? 

setOnClickListener(new OnClickListener() {
   @Override
    public void onClick(View v){
       ...
   }
}

Trying to think what would be easier for students to understand.

Reply:
I think a lambda should be fine to use here. Is there an advantage to the method you've suggested?

The reasons I used a lambda here are:

1. It's short and concise.
2. The IDE will suggest replacing it with a lambda. (See image below.)
3. Lambdas are used in Kotlin, as well.
4. It's a Java 8 feature supported by the Gradle plugin.

<img width="732" alt="image" src="https://user-images.githubusercontent.com/104776914/188701801-2a79e993-479e-4785-8dd5-b1645a008a17.png">

Reply:
Ah got it, okay sounds good. In that case, could you add a quick mention in the codelab about the long-form way and then say how it can be expressed with a lambda?

Reply:
This is covered in the codelab. 👍